python/pyarrow/tests/test_schema.py

```diff
@@ -69,6 +69,47 @@ def test_type_list():
     assert str(l2) == 'list<my_item: string>'
 
 
+def test_type_for_alias():
+    cases = [
+        ('i1', pa.int8()),
```

Comment:
I personally am not a huge fan of the `i*` and `u*` numpy aliases because they aren't very explicit and they use different units than their more verbose counterparts. NumPy also has `u8` and `U8` which mean completely different things. I guess for maximum compatibility they are useful, but not for much else.

Reply:
agreed. We should use the longhand versions in any real code